keps/sig-instrumentation/20191115-structured-logging.md

```diff
@@ -0,0 +1,424 @@
+* Will enabling / disabling the feature require downtime for the control plane? **Yes, changing flag passed will require restarting components**
+* Will enabling / disabling the feature require downtime or reprovisioning of a node? **Yes, if we are changing logging configuration of kubelet will require node downtime**
+* What happens if a cluster with this feature enabled is rolled back? What happens if it is subsequently upgraded again? **Temporary change in log format produced**
+* Are there tests for this? **Klog implementation is covered by integration tests**
+
+**Scalability**
+* Will enabling / using the feature result in any new API calls? Describe them with their impact keeping in mind the supported limits (e.g. 5000 nodes per cluster, 100 pods/s churn): **No**
+* Will enabling / using the feature result in supporting new API types? How many objects of that type will be supported (and how that translates to limitations for users)? **No**
+* Will enabling / using the feature result in increasing size or count of the existing API objects? **No**
+* Will enabling / using the feature result in increasing time taken by any operations covered by existing SLIs/SLOs (e.g. by adding additional work, introducing new steps in between, etc.)? **No**
+* Will enabling / using the feature result in non-negligible increase of resource usage (CPU, RAM, disk IO, ...) in any components? Things to keep in mind include: additional in-memory state, additional non-trivial computations, excessive access to disks (including increased log volume), significant amount of data sent and/or received over network, etc. Think through this in both small and large cases, again with respect to the supported limits. **Yes, increased log volume increase below 50%**
+* Rollout, Upgrade, and Rollback Planning
+* Dependencies
+    * Does this feature depend on any specific services running in the cluster (e.g., a metrics service)? **No**
+    * How does this feature respond to complete failures of the services on which it depends? **n/a**
+    * How does this feature respond to degraded performance or high error rates from services on which it depends? **n/a**
+* Monitoring requirements
+    * How can an operator determine if the feature is in use by workloads? **Need to verify specific component flag, or look into logs generated by it**
+    * How can an operator determine if the feature is functioning properly? **By looking at the logs generated by component**
+    * What are the service level indicators an operator can use to determine the health of the service? **n/a**
+    * What are reasonable service level objectives for the feature? **Defining a proper SLO for logging should consider e2e delivery to backend. This would be outside of scope of kubernetes**
+* Troubleshooting
+    * What are the known failure modes? **Logging is a blocking API. This means that in case of full node disk, logging can halt the process**
```

Comment:
Only Error/Fatal is blocking, right?

Reply:
I don't think so. I didn't find difference in Error/Fatal logs are written. Best to ask @thockin

Reply:
in klog, I think that is correct.